janus_core/processing/observables.py

```diff
@@ -1,72 +1,307 @@
+        include_ideal_gas: bool = True,
+    ):
         """
-        Initialise the observables from a symbolic str component.
+        Initialise the observable from a symbolic str component.
 
         Parameters
         ----------
-        component : str
-            Symbol for tensor components, xx, yy, etc.
+        components : list[str]
+            Symbols for tensor components, xx, yy, etc.
+        atoms_slice : list[int] | SliceLike | None = None
+            List or slice of atoms to observe velocities from.
         include_ideal_gas : bool
             Calculate with the ideal gas contribution.
         """
-        components = {
-            "xx": 0,
-            "yy": 1,
-            "zz": 2,
-            "yz": 3,
-            "zy": 3,
-            "xz": 4,
-            "zx": 4,
-            "xy": 5,
-            "yx": 5,
-        }
-        if component not in components:
-            raise ValueError(
-                f"'{component}' invalid, must be '{', '.join(list(components.keys()))}'"
-            )
+        ComponentMixin.__init__(
+            self,
+            components={
+                "xx": 0,
+                "yy": 1,
+                "zz": 2,
+                "yz": 3,
+                "zy": 3,
+                "xz": 4,
+                "zx": 4,
+                "xy": 5,
+                "yx": 5,
+            },
+        )
+        self._set_components(components)
+
+        if atoms_slice:
+            self.atoms_slice = atoms_slice
+        else:
+            self.atoms_slice = slice(0, None, 1)
 
-        self.component = component
-        self._index = components[self.component]
+        Observable.__init__(self, len(components))
         self.include_ideal_gas = include_ideal_gas
 
-    def __call__(self, atoms: Atoms, *args, **kwargs) -> float:
+    def __call__(self, atoms: Atoms) -> list[float]:
         """
-        Get the stress component.
+        Get the stress components.
 
         Parameters
         ----------
         atoms : Atoms
             Atoms object to extract values from.
-        *args : tuple
-            Additional positional arguments passed to getter.
-        **kwargs : dict
-            Additional kwargs passed getter.
 
         Returns
         -------
-        float
-            The stress component in GPa units.
+        list[float]
+            The stress components in GPa units.
         """
+        sliced_atoms = atoms[self.atoms_slice]
+        sliced_atoms.calc = atoms.calc
         return (
-            atoms.get_stress(include_ideal_gas=self.include_ideal_gas, voigt=True)[
-                self._index
-            ]
+            sliced_atoms.get_stress(
+                include_ideal_gas=self.include_ideal_gas, voigt=True
+            )
             / units.GPa
-        )
+        )[self._indices]
+
+
+StressDiagonal = Stress(components=["xx", "yy", "zz"])
+ShearStress = Stress(components=["xy", "yz", "zx"])
```

Comment:
Don't like that this isn't `StressShear` to go with others. May also want `StressHydrostatic` rather than `Diagonal`?

Reply:
those are both good ideas,```StressHydrostatic``` sounds good, unless someone can think of better I'll go with it